src/rust/engine/src/context.rs

```diff
@@ -185,45 +180,8 @@ impl Core {
     }
   }
 
-  pub fn fork_context<F, T>(&self, f: F) -> T
```

Comment:
I think that this is correct to remove, like we discussed.

I would mention this in the PR description and title, though, that this PR does a little bit more than the issue discusses. Maybe tweak the title to something more general like "Remove fork-related code from Rust".